### ISSUE TITLE
Add setting for permission_classes for views

### ISSUE THREAD

**Issue by brycekellogg** [James1345/django-rest-knox#201]
Currently the `LoginView`, `LogoutView`, and `LogoutAllView` all have hardcoded `permission_classes` to `IsAuthenticated`. While this is definitely the minimum needed as discussed in issues #64 and #9, sometimes additional permissions are required. For our use case, we are using API keys and only want a user to be able to login/out/etc if they have an API key. Currently we solve this like so:
```python
class LoginView(KnoxLoginView):
    permission_classes = (IsAuthenticated,HasAPIKey)

class LogoutView(KnoxLogoutView):
    permission_classes = (IsAuthenticated,HasAPIKey)

class LogoutAllView(KnoxLogoutAllView):
    permission_classes = (IsAuthenticated,HasAPIKey)
```
One possible solution could be similar to how Djoser does it:

```python
class UserViewSet(viewsets.ModelViewSet):
    permission_classes = settings.PERMISSIONS.user
```

where we set the permissions in settings.py like so:
```python
DJOSER = {
    'PERMISSIONS': {
        'user':  ['baas.auth.permissions.HasAPIKey', 'djoser.permissions.CurrentUserOrAdmin'],
    }
}
```

**Comment by johnraz** [comment]
I agree this is a valid option but introducing a new settings means new documentation, more complexity in the code base and more tests. As this is a small project with very limited resources I think we should keep it as is.
